sound/soc/intel/boards/bdw-rt5677.c

```diff
@@ -282,6 +282,15 @@ static int bdw_rt5677_init(struct snd_soc_pcm_runtime *rtd)
 	return 0;
 }
 
+static void bdw_rt5677_exit(struct snd_soc_pcm_runtime *rtd)
+{
+	struct bdw_rt5677_priv *bdw_rt5677 =
+			snd_soc_card_get_drvdata(rtd->card);
+
+	if (!IS_ERR(bdw_rt5677->gpio_hp_en))
+		gpiod_put(bdw_rt5677->gpio_hp_en);
```

Comment:
This function is called from a nested loop: the external loop is over "component orders," the internal loop is over RTDs, but after looking through both of them it looks like this function will provably only be called once. So, it should be safe to just do as in this patch without setting `bdw_rt5677->gpio_hp_en` to an error code to prevent a repeated `gpiod_put()` call. But if there's any doubt, that this will stay like this (e.g. if another DAI link will be added with the same `.init()` and `.exit()`?) maybe it's better to set the pointer to `-EINVAL`

Reply:
Thanks @lyakh indeed I missed the outer loop, which is yet another reason to move the code to soc_free_pcm_runtime.